### PR TITLE
Add nice stop method to VMEventManager

### PR DESCRIPTION
**Pull request by Ma1eusz**
Adding gentle stop for VMEventManager as well as making this class not to extend Thread class. 
See issue #1373

### CODE REVIEW

File: core/src/main/scala/org/ensime/core/debug/VMEventManager.scala [hunk 1]
```diff
@@ -7,37 +7,43 @@ import com.sun.jdi.VMDisconnectedException
...
-        val it = eventSet.eventIterator()
-        while (it.hasNext) {
-          val evt = it.nextEvent()
-          evt match {
-            case e: VMDisconnectEvent =>
-              finished = true
-            case e: ClassPrepareEvent =>
-              debugManager ! DMClassPrepareEvent(e, eventSet)
-            case _ =>
+  val thread = new Thread() {
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/core/debug/VMEventManager.scala]: I would make this private

File: core/src/main/scala/org/ensime/core/debug/VMEventManager.scala [hunk 2]
```diff
@@ -7,37 +7,43 @@ import com.sun.jdi.VMDisconnectedException
...
+    override def run(): Unit = {
+      while (!this.isInterrupted) {
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/core/debug/VMEventManager.scala]: I think I would actually keep the finished flag - and then the stop method sets the flag and interrupts.  - that means the other catches just set log something and set the flag.  Feels cleaner to me - but it is stylistic.

**Comment by rorygraves** [comment]
Thanks for the PR.  Can you please update the commit message to include the brief change description?

**Comment by fommil** [comment]
can you please rebase?

**Comment by Ma1eusz** [comment]
@fommil Done

**Comment by chipsenkbeil** [comment]
@fommil shouldn't this have a debugger label on it since it's touching the existing debugger code?

**Comment by fommil** [comment]
I don't tend to add labels to PRs, it's too much effort

**Comment by fommil** [comment]
Thank you!

**Pull request merged**